### PR TITLE
Added CRC tenet to check Context being passed as first argument

### PR DESCRIPTION
**Pull request by Daanikus**
Tenet for https://github.com/golang/go/wiki/CodeReviewComments#contexts

### CODE REVIEW

File: tenets/codelingo/code-review-comments/context-first-arg/codelingo.yaml [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+        title: Context as First Argument
+        body: |
+          Values of the context.Context type carry security credentials, tracing information, 
+          deadlines, and cancellation signals across API and process boundaries. Go programs 
+          pass Contexts explicitly along the entire function call chain from incoming RPCs 
+          and HTTP requests to outgoing requests.
+
+          Most functions that use a Context should accept it as their first parameter.
+      codelingo/review:
+        comment: Most functions that use a Context should accept it as their first parameter.
```
Comment (mullikine) [inline tenets/codelingo/code-review-comments/context-first-arg/codelingo.yaml]: The comment's basically fine. But it has to tell the user that this is being done wrong. Maybe change to this "Most functions that use a Context should accept it as their first parameter. Consider moving `context` to the front."

File: tenets/codelingo/code-review-comments/context-first-arg/codelingo.yaml [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+    query: |
+      import codelingo/ast/go
+
+      go.field_list(depth = any):
```
Comment (mullikine) [inline tenets/codelingo/code-review-comments/context-first-arg/codelingo.yaml]: field_list could also be the list of return parameters. Unfortunately there is no easy way around this. put a "TODO use macro fact to ensure this is not a return parameter"

File: tenets/codelingo/code-review-comments/context-first-arg/expected.json [hunk 3]
```diff
@@ -0,0 +1,14 @@
+[
```
Comment (mullikine) [inline tenets/codelingo/code-review-comments/context-first-arg/expected.json]: I'm getting different output for expected.json. Now I'm questioning my own methodology. Must ask blake / jess / reuben which directory this should be run from to ensure unit tests pass

```
diff --git a/expected.json b/expected.json
index 9719649..3171cba 100755
--- a/expected.json
+++ b/expected.json
@@ -1,14 +1,14 @@
 [
   {
    "Comment": "Most functions that use a Context should accept it as their first parameter.",
-   "Filename": "effective-go/context-first-arg/example.go",
+   "Filename": "example.go",
    "Line": 9,
-   "Snippet": "func F(ctx context.Context, a int)               {}\nfunc F(b int, ctx context.Context, a int)        {}\nfunc F(c int, b int, ctx context.Context, a int) {}\n"
+   "Snippet": "\nfunc F(ctx context.Context, a int)               {}\nfunc F(b int, ctx context.Context, a int)        {}\nfunc F(c int, b int, ctx context.Context, a int) {}"
   },
   {
    "Comment": "Most functions that use a Context should accept it as their first parameter.",
-   "Filename": "effective-go/context-first-arg/example.go",
-   "Line": 8,
-   "Snippet": "\nfunc F(ctx context.Context, a int)               {}\nfunc F(b int, ctx context.Context, a int)        {}\nfunc F(c int, b int, ctx context.Context, a int) {}"
+   "Filename": "example.go",
+   "Line": 10,
+   "Snippet": "func F(ctx context.Context, a int)               {}\nfunc F(b int, ctx context.Context, a int)        {}\nfunc F(c int, b int, ctx context.Context, a int) {}\n"
   }
  ]
\ No newline at end of file
```

File: tenets/codelingo/code-review-comments/context-first-arg/example.go [hunk 4]
```diff
@@ -0,0 +1,12 @@
...
+
+import (
+	"fmt"
+)
+
+func a(ctx context.Context, a int)               {}
+
+func b(b int, ctx context.Context, a int)        {}
+
+func c(c int, b int, ctx context.Context, a int) {}
```
[the remark this answers is not in the document]
Comment (Daanikus) [reply tenets/codelingo/code-review-comments/context-first-arg/example.go]: /ignore

File: tenets/codelingo/code-review-comments/context-first-arg/example.go [hunk 5]
```diff
@@ -0,0 +1,12 @@
+// Package main used for tenet testing
+package main
+
+import (
+	"fmt"
+)
+
+func a(ctx context.Context, a int)               {}
+
+func b(b int, ctx context.Context, a int)        {}
```
[the remark this answers is not in the document]
Comment (Daanikus) [reply tenets/codelingo/code-review-comments/context-first-arg/example.go]: /ignore

**Pull request merged**